src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -3081,19 +3081,6 @@ ErrorIfUnsupportedTableCombination(Query *queryTree)
 		}
 	}
 
-	if ((subqueryRangeTableCount > 0) && (relationRangeTableCount > 0))
```

Comment:
done, for the first part.
I think this current function name is still good to use.